lib/features/appConfiguration/data/remote_app_configuration_repository.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:zachranobed/features/appConfiguration/data/entity/remote_app_configuration_dto.dart';
+
+class RemoteAppConfigurationRepository {
+  final _collection = FirebaseFirestore.instance.collection('appConfiguration').doc('app-terms').withConverter(
+      fromFirestore: (snapshot, _) {
+        final json = snapshot.data() ?? {};
+        json['id'] = snapshot.id;
+
+        // Mapping to correct property name (`RemoteAppConfigurationDto.lastAppTermsVersion`)
+        json['lastAppTermsVersion'] = snapshot['lastVersion'];
+
+        return RemoteAppConfigurationDto.fromJson(json);
```

Comment:
Why this remapping? `RemoteAppConfigurationDto` is a DTO, so you could name it as in firestore

Also I guess `json['id'] = snapshot.id;` is not needed, as DTO has no "id" field to map to

Reply:
I have to disagree with this one since we store the value in firebase in nested context `app-terms.lastVersion`. In the context of the DTO, we have parameters that are related to remote app configuration, but not specific nested parts - that means that I prefere to use `lastAppTermsVersion ` rather than `lastVersion ` which would not provide enough context for me

I removed the id since we dont need it